src/azul/plugins/metadata/hca/transform.py

```diff
@@ -840,39 +845,72 @@ def _sequencing_input(self, sequencing_input: api.Biomaterial) -> MutableJSON:
     @classmethod
     def _sample_types(cls) -> FieldTypes:
         return {
+            'document_id': null_str,
+            'biomaterial_id': null_str,
             'entity_type': null_str,
+            'organ': null_str,
+            'organ_part': null_str,
+            'model_organ': null_str,
+            'model_organ_part': null_str,
             'effective_organ': null_str,
-            **cls._cell_line_types(),
-            **cls._organoid_types(),
-            **cls._specimen_types()
         }
 
-    def _sample(self, sample: api.Biomaterial) -> MutableJSON:
-        # Start construction of a `sample` inner entity by including all fields
-        # possible from any entities that can be a sample. This is done to
-        # have consistency of fields between various sample inner entities
-        # to allow Elasticsearch to search and sort against these entities.
-        sample_ = dict.fromkeys(ChainMap(
-            self._cell_line_types(),
-            self._organoid_types(),
-            self._specimen_types()
-        ).keys())
-        entity_type, entity_dict = (
-            'cell_lines', self._cell_line(sample)
-        ) if isinstance(sample, api.CellLine) else (
-            'organoids', self._organoid(sample)
-        ) if isinstance(sample, api.Organoid) else (
-            'specimens', self._specimen(sample)
-        ) if isinstance(sample, api.SpecimenFromOrganism) else (
-            require(False, sample), None
-        )
-        sample_.update(entity_dict)
-        sample_['entity_type'] = entity_type
-        assert hasattr(sample, 'organ') != hasattr(sample, 'model_organ')
-        sample_['effective_organ'] = sample.organ if hasattr(sample, 'organ') else sample.model_organ
-        assert sample_['document_id'] == str(sample.document_id)
-        assert sample_['biomaterial_id'] == sample.biomaterial_id
-        return sample_
+    def _samples(self, samples: List[api.Biomaterial]) -> Mapping[str, JSONs]:
+        """
+        Returns inner entity values with all given samples represented both in a
+        'samples' inner entity and a type-specific 'sample_{entity_type}'.
+        The 'samples' inner entity is a minified polymorphic field containing
+        the few properties common to all samples. This allows facets on these
+        properties regardless of the sample entity type.
+        """
+
+        def sample_cell_line(entity):
```

Comment:
see if making these methods and moving closer to their sibling (like _cell_line) makes sense

Reply:
Not sure if this better or worse for organization than keeping these as inner functions of `_samples`.
I implemented this a a separate fixup https://github.com/DataBiosphere/azul/pull/2740/commits/d82801c38137ee950ea648fd8a5ccc49cf9bf901 so it could be compared.